test/jdk/javax/accessibility/JEditorPane/TestEditorPaneAccessibleChildCount.java

```diff
@@ -0,0 +1,130 @@
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8273986
+ * @key headful
+ * @requires (os.family == "windows")
+ * @summary Verifies if accessible child count for JEditorPane HTML demo
+ * returns correct child count.
+ * @run main TestEditorPaneAccessibleChildCount
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Dimension;
+import java.io.IOException;
+import java.net.URL;
+import javax.accessibility.AccessibleContext;
+import javax.swing.JEditorPane;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.SwingUtilities;
+import javax.swing.text.html.HTMLEditorKit;
+
+public class TestEditorPaneAccessibleChildCount {
+    private static JEditorPane jep;
+    private static AccessibleContext ac;
+    private static JFrame frame;
+
+    public TestEditorPaneAccessibleChildCount() {
+        createAndShowUI();
+    }
+
+    public void createAndShowUI() {
+        frame = new JFrame("JEditorPane A11Y Child Count Test");
+        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        JPanel panel = new JPanel();
+        jep = new JEditorPane();
+        jep.setEditable(false);
+        jep.setEditorKit(new HTMLEditorKit());
+        URL url = TestEditorPaneAccessibleChildCount.class.
+                getResource("test1.html");
+        loadHtmlPage(url);
+        JScrollPane jScrollPane = new JScrollPane(jep);
+        jScrollPane.setPreferredSize(new Dimension(540,400));
+        panel.add(jScrollPane);
+        frame.getContentPane().add(panel, BorderLayout.CENTER);
+        frame.setSize(560, 450);
+        frame.setLocationRelativeTo(null);
+        frame.pack();
+        frame.setVisible(true);
+    }
+
+    public static void loadHtmlPage(URL url) {
+        try {
+            jep.setPage(url);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static void addDelay(int mSec) {
+        try {
+            Thread.sleep(mSec);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
+        try {
+            SwingUtilities.invokeAndWait(() -> {
+                new TestEditorPaneAccessibleChildCount();
+            });
+
+            addDelay(500);
+            SwingUtilities.invokeAndWait(() -> {
+                ac = jep.getAccessibleContext();
+            });
+            int childCount1 = ac.getAccessibleChildrenCount();
+
+            URL url = TestEditorPaneAccessibleChildCount.class.
+                        getResource("test2.html");
+            SwingUtilities.invokeAndWait(() -> {
+                loadHtmlPage(url);
+            });
+            addDelay(500);
+            int childCount2 = ac.getAccessibleChildrenCount();
```

Comment:
note: "ac" initialized on EDT, and accessed here w/o synchronization.

Reply:
I am facing an error `local variables referenced from a lambda expression must be final or effectively final` when I tried to get the ChildrenCount and assign it to local variable `childCount2` on EDT.

My piece of code looks like this in `main` method

```
int childCount2 = 0;
SwingUtilities.invokeAndWait(() -> {
        childCount2 = ac.getAccessibleChildrenCount();
 });
```

Reply:
@mrserb Accessing the `accessibleContext (ac)` on EDT now. Moved the local variables (`childCount1 and childCount2`) to class variable and assigned the values on EDT.

Please review the changes.

Reply:
@mrserb Please review the latest changes.